### PR TITLE
Add sensitive form widget

### PR DESCRIPTION
**Pull request by tomaszstrojny**
As discussed in the https://github.com/octobercms/october/issues/5061 I think it is a good idea to introduce a new form widget, `sensitive`:

passwd

In this PR I am changing the field type for SMTP password in the backend as well.

If you like the idea and the code I will create a new PR for the documentation.

### CODE REVIEW

File: modules/backend/formwidgets/Sensitive.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use ApplicationException;
+use Backend\Classes\FormWidgetBase;
+
+/**
+ * Sensitive - renders a password field with "eye" icon to show it
+ */
+class Sensitive extends FormWidgetBase
+{
+    /**
+     * @var bool If true, the color picker is set to read-only mode
```
Comment (bennothommo) [inline modules/backend/formwidgets/Sensitive.php:13]:
```suggestion
     * @var bool If true, the sensitive field is set to read-only mode
```

File: modules/backend/formwidgets/Sensitive.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    public $readOnly = false;
+
+    /**
+     * @var bool If true, the color picker is set to disabled mode
```
Comment (bennothommo) [inline modules/backend/formwidgets/Sensitive.php:18]:
```suggestion
     * @var bool If true, the sensitive field is set to disabled mode
```

File: modules/backend/formwidgets/sensitive/assets/js/sensitive.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            self.togglePassword()
+            self.toggleIcon()
+        })
+    }
+
+    Sensitive.prototype.togglePassword = function() {
+        if (this.$sensitiveField.attr("type") == "password") {
+            this.$sensitiveField.attr("type", "text");
+        } else {
+            this.$sensitiveField.attr("type", "password");
```
Comment (bennothommo) [inline modules/backend/formwidgets/sensitive/assets/js/sensitive.js:28]: We prefer strings to be defined with single quotes.

```suggestion
        if (this.$sensitiveField.attr('type') === 'password') {
            this.$sensitiveField.attr('type', 'text');
        } else {
            this.$sensitiveField.attr('type', 'password');
```

File: modules/backend/formwidgets/sensitive/assets/js/sensitive.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+
+    Sensitive.prototype.toggleIcon = function() {
+        this.$showIcon.find('i').toggleClass("icon-eye icon-eye-slash");
```
Comment (bennothommo) [inline modules/backend/formwidgets/sensitive/assets/js/sensitive.js:33]:
```suggestion
        this.$showIcon.find('i').toggleClass('icon-eye icon-eye-slash');
```

File: modules/backend/formwidgets/sensitive/assets/js/sensitive.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            if (typeof result != 'undefined') return false
+        })
+
+        return result ? result : this
+    }
+
+    $.fn.sensitive.Constructor = Sensitive
+
+    $.fn.sensitive.noConflict = function () {
+        $.fn.sensitive = old
```
Comment (bennothommo) [inline modules/backend/formwidgets/sensitive/assets/js/sensitive.js:53]: `old` is not defined anywhere.

**Comment by tomaszstrojny** [comment]
Okay, I hope it's okay right now. I'm writing JS totally blindfolded so doublecheck it, please :)

I've added the dispose method according to the document you have sent, thanks for it :)

File: modules/backend/formwidgets/sensitive/partials/_sensitive.htm [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        type="password"
+        data-control="sensitive"
+        name="<?= $this->getFieldName() ?>"
+        id="<?= $this->getId() ?>"
+        value="<?= e($this->getLoadValue()) ?>"
+        placeholder="<?= e(trans($this->placeholder)) ?>"
+        class="form-control"
+        <?php if ($this->previewMode): ?>
+        disabled="disabled"
+        <?php endif ?>
```
Comment (LukeTowers) [inline modules/backend/formwidgets/sensitive/partials/_sensitive.htm:13]: make this one line please, the preview mode check

**Comment by tomaszstrojny** [comment]
I have found another bug while making this one-liner. Hope it's okay now :)

**Comment by bennothommo** [comment]
@github-actions run

<details>
<summary>✔ Ready for testing</summary>

```js
(async () => {
  // Get issue/PR number
  const number = context.issue.number;

  // Add Testing Needed label
  await githubClient.issues.addLabels({
    owner: context.repo.owner,
    repo: context.repo.repo,
    issue_number: number,
    labels: ['Testing Needed'],
  });

  // Comment body
  const commentBody = 'This pull request has been marked as **Ready for Testing**. To expedite this pull request being'
    + ' merged, we encourage members of the community to test the changes and confirm they are working. You may view our'
    + ' [Testing a Pull Request](https://github.com/octobercms/october/blob/develop/.github/CONTRIBUTING.md#testing-a-pull-request)'
    + ' section in the Contibution Guide to access the changes in your own copy of October CMS.'
    + '\n\n'
    + 'If you have [Docker](https://www.docker.com/), you can quickly run a copy of October CMS with this pull request\'s'
    + ' changes in place via the following command:'
    + '\n\n'
    + '```\n'
    + 'docker run --rm -p 80:80 -e GIT_MERGE_PR=' + number + ' aspendigital/octobercms:develop\n'
    + '```';

  // Add comment
  await postComment(commentBody);
})();
```

File: modules/backend/lang/pl/lang.php [hunk 7]
```diff
@@ -1,568 +1,627 @@
...
+        'selection_mode_fixed_size' => 'Zablokuj rozmiar',
+        'height' => 'Wysokość',
+        'width' => 'Szerokość',
+        'selection_mode' => 'Tryb zaznaczenia',
+        'resize_image' => 'Zmień rozmiar obrazka',
+        'image_size' => 'Rozmiar obrazka:',
+        'selected_size' => 'Wybrano:',
     ],
-];
+];
```
Comment (bennothommo) [inline modules/backend/lang/pl/lang.php:627]: @tomaszstrojny Our code checks required the last line of each PHP file to be an empty line.
```suggestion
];

```

Comment (tomaszstrojny) [reply modules/backend/lang/pl/lang.php:627]: Nah, sorry about that, I have pushed the translations from a different PR to this branch... https://github.com/octobercms/october/pull/5129

**Comment by bennothommo** [comment]
@tomaszstrojny We're going to hold off merging this until 468 - 467 is a bug-fix release only. We'll still need to test it first though :)

**Comment by bennothommo** [comment]
@tomaszstrojny I've opened up #5201 which uses your initial implementation, and adds a few ideas of my own. I've made sure to credit you for the initial implementation. Feel free to let me know your thoughts on there.

**Pull request closed without merging**